swim/node.go

```diff
@@ -133,13 +145,17 @@ type Node struct {
 		sync.RWMutex
 	}
 
-	channel      shared.SubChannel
-	memberlist   *memberlist
-	memberiter   memberIter
-	disseminator *disseminator
-	suspicion    *suspicion
-	gossip       *gossip
-	rollup       *updateRollup
+	channel          shared.SubChannel
+	discoverProvider discovery.DiscoverProvider
+	memberlist       *memberlist
+	memberiter       memberIter
+	disseminator     *disseminator
+	suspicion        *suspicion
+	gossip           *gossip
+	rollup           *updateRollup
```

Comment:
You added only one field here -- `discoverProvider`. Maybe you can add it to the end of the list, so diff is smaller?